Source/UI/NewUI/CommonUI/View/TableViewCell/CKOCellButton.swift

```diff
@@ -0,0 +1,42 @@
+import UIKit
+
+protocol CKOCellButtonDelegate: AnyObject {
+    func buttonIsPressed()
+}
+
+final class CKOCellButton: UITableViewCell {
+    weak var delegate: CKOCellButtonDelegate?
+
+    private var mainView: UIView
+    init(mainView: UIView) {
+        self.mainView = mainView
+        super.init(style: .default, reuseIdentifier: nil)
+        self.setupViewsInOrder()
+    }
+
+    @available(*, unavailable)
+    required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+extension CKOCellButton {
+
+    private func setupViewsInOrder() {
+        contentView.addSubview(mainView)
+        NSLayoutConstraint.activate([
+            mainView.topAnchor.constraint(
+                equalTo: contentView.topAnchor),
+            mainView.leadingAnchor.constraint(
+                equalTo: contentView.leadingAnchor),
+            mainView.trailingAnchor.constraint(
+                equalTo: contentView.trailingAnchor),
+            mainView.bottomAnchor.constraint(
```

Comment:
If an extension on UIView for adding subviews with automatic paddings doesn't already exist I would create it:
```
extension UIView {

    func addContainedSubview(_ subview: UIView, with insets: UIEdgeInsets = .zero) {
        addSubview(subview)

        NSLayoutConstraint.activate([
            topAnchor.constraint(equalTo: subview.topAnchor, constant: insets.top),
            leadingAnchor.constraint(equalTo: subview.leadingAnchor, constant: insets.left),
            subview.bottomAnchor.constraint(equalTo: bottomAnchor, constant: insets.bottom),
            subview.trailingAnchor.constraint(equalTo: trailingAnchor, constant: insets.right)
        ])
    }

}
```

This can then be fully unit tested to ensure it behaves as intended.